index.rst

```diff
@@ -53,9 +53,77 @@
+This makes it almost impossible for the verification and validation team to read more than a few hundred into an analysis notebook.
+
+Butler Datasets
+===============
+
+Every metric generated in a pipeline is treated as a butler dataset.
+A butler dataset has a UUID that can uniquely identify it, but it also is guaranteed to be locatable using:
+
+* A RUN collection.
+* A dataset type.
+* A data coordinate.
+
+A metrics database that contains datasets from a butler repository therefore needs to include the run and data coordinate as well as the dataset type.
+The data coordinate allows the metric to be located in time or space, and the time-based lookup can be used to associate the metric with other EFD telemetry.
+
+Furthermore, butler has the concept of CHAINED collections where a dataset can be located by returning the first match in the chain.
+How CHAINED collections map to a metrics database outside of butler control is not yet determined.
+
+Ways Forward
+============
+
+There are some ways by which the metrics tracking system can be improved.
+
+* Write collation pipeline tasks that will run in the batch system.
+  These tasks will collect related metrics and store them in a single butler dataset, possibly in parquet format.
+* Transfer metrics into the SQuaSH system :cite:`SQR-009` as an afterburner (possibly by reading the collated parquet files).
+* Create a Sasquatch/SQuaSH :cite:`SQR-068` datastore backend that can be used directly from batch workflows to stream metrics into the EFD (whilst also storing them as individual datasets using a chained datastore).
+
+Sasquatch
+=========
+
+The Sasquatch system :cite:`SQR-068` is a metrics-tracking system fully integrated into the EFD.
+The current design allows for metrics to be loaded that can track the dataset type, processing run, and dataId.
+Sasquatch is a time-series database and has been used by the Data Management teacm to track how metrics evolve with changes to software versions.
+
+In the context of an operational observatory, it will be necessary for metrics to be associated with the time of observation so that it will be possible to correlate telemetry from the EFD with metrics (for example plotting measured seeing against wind speed).
+
+Each (exposure-based) metric stored in the butler repository will have two useful timestamps.
+The first is the timestamp of the original observation and the second is the time the metric was calculated.
+Additionally, the software version and other provenance information can be useful.
+
+Sasquatch does not currently support two timestamps but InfluxDB 2.0 can support this.
+The system can not convert a ``visit`` or ``exposure`` entry into a time, so the code that migrates Butler metrics into Sasquatch must extract the relevant time-of-observation from the dimension record attached to the data coordinate.
```

Comment:
Sasquatch should not be responsible for finding the time of observation for a ``visit`` or ``exposure``, this timestamp will be obtained by the code that sends the Butler metrics to Sasquatch and added as metadata.
Sasquatch can use that timestamp to index the metrics database to allow correlations with the EFD data.
In addition, the timestamp of the metric calculation is going to be recorded as another field in InfluxDB and can also be used as the time axis for visualizations in Chronograf dashboards if needed. This is already possible using the `Flux query language`_ in InfluxDB 1.x.
InfluxDB 2.x supports new `visualization types`_ like scatter plots, histograms, and heatmaps that can also be useful.  

.. _visualization types:: https://docs.influxdata.com/influxdb/latest/visualize-data/visualization-types/
.. _Flux query language:: https://docs.influxdata.com/flux/latest/get-started/